mp/src/game/client/neo/ui/neo_scoreboard.h

```diff
@@ -4,43 +4,118 @@
+	vgui::VPANEL GetVPanel( void ) { return BaseClass::GetVPanel(); }
+  	virtual bool IsVisible() { return BaseClass::IsVisible(); }
+  	virtual void SetParent( vgui::VPANEL parent ) { BaseClass::SetParent( parent ); }
+
+	// IGameEventListener interface:
+	virtual void FireGameEvent( IGameEvent *event);
+
+	virtual void UpdatePlayerAvatar( int playerIndex, KeyValues *kv );
+
 protected:
-	// scoreboard overrides
-	virtual void InitScoreboardSections() OVERRIDE;
-	virtual void UpdatePlayerInfo() OVERRIDE;
-	virtual void UpdateTeamInfo() OVERRIDE;
-	virtual bool GetPlayerScoreInfo(int playerIndex, KeyValues *outPlayerInfo) OVERRIDE;
-	virtual void SetVisible(bool state) OVERRIDE;
+	MESSAGE_FUNC_INT( OnPollHideCode, "PollHideCode", code );
+
+	// functions to override
+	virtual void GetPlayerScoreInfo(int playerIndex, KeyValues *outPlayerInfo);
+	virtual void InitScoreboardSections();
+	virtual void UpdateTeamInfo();
+	virtual void UpdatePlayerInfo();
+	virtual void OnThink();
+	virtual void AddSection(int teamType, int teamNumber); // add a new section header for a team
+	virtual int GetAdditionalHeight() { return 12; }
 
-	// vgui overrides
-	virtual void PaintBackground() OVERRIDE;
-	virtual void PaintBorder() OVERRIDE;
-	virtual void ApplySchemeSettings(vgui::IScheme *pScheme) OVERRIDE;
+	// sorts players within a section
+	static bool StaticPlayerSortFunc(vgui::SectionedListPanel *list, int itemID1, int itemID2);
 
-	bool IsTeamplay(void) { return true; }
+	virtual void ApplySchemeSettings(vgui::IScheme *pScheme);
+
+	virtual void PostApplySchemeSettings( vgui::IScheme *pScheme );
+
+	// finds the player in the scoreboard
+	int FindItemIDForPlayerIndex(vgui::SectionedListPanel *pPlayerList, int playerIndex);
+
+	int m_iNumTeams;
+
+	vgui::SectionedListPanel *m_pJinraiPlayerList;
+	vgui::SectionedListPanel *m_pNSFPlayerList;
+	vgui::SectionedListPanel *m_pSpectatorsPlayerList;
+
+	int s_VoiceImage[5];
+	int TrackerImage;
+	int	m_HLTVSpectators;
+	int m_ReplaySpectators;
+	int	m_iDesiredHeight;
+	float m_fNextUpdateTime;
+
+	void MoveLabelToFront(const char *textEntryName);
+	void MoveToCenterOfScreen();
+
+	vgui::ImageList				*m_pImageList;
+	CUtlMap<CSteamID,int>		m_mapAvatarsToImageList;
+
+	CPanelAnimationVarAliasType( int, m_iPingWidth, "ping_width", "25", "proportional_int" );
+	CPanelAnimationVar( int, m_iAvatarWidth, "avatar_width", "40" );		// Avatar width doesn't scale with resolution
+	CPanelAnimationVarAliasType( int, m_iNameWidth, "name_width", "110", "proportional_int" );
+	CPanelAnimationVar(int, m_iStatusWidth, "status_width", "40");
+	CPanelAnimationVarAliasType( int, m_iClassWidth, "class_width", "40", "proportional_int" );
+	CPanelAnimationVarAliasType(int, m_iRankWidth, "rank_width", "60", "proportional_int");
+	CPanelAnimationVarAliasType( int, m_iScoreWidth, "score_width", "25", "proportional_int" );
+	CPanelAnimationVarAliasType( int, m_iDeathWidth, "death_width", "25", "proportional_int" );
```

Comment:
Should be `deaths_width`

Reply:
I made everything singular, same as the original scoreboard.